ketcher-autotests/tests/Templates/Functional-Groups/functional-groups.spec.ts

```diff
@@ -645,4 +678,44 @@ test.describe('Functional Groups', () => {
 
     await page.keyboard.press('n');
   });
+
+  test('Add a custom structure to a canvas with an expanded functional group and contract it', async ({
+    page,
+  }) => {
+    /*
+    Test case: EPMLSOPKET-12978
+    Description: Functional Group contract and remove abbreviation
+    */
+    await openFileAndAddToCanvas(
+      'Molfiles-V2000/custom-structure-with-expanded-fg.mol',
+      page,
+    );
+
+    await page.getByText('CO2Et').click({ button: 'right' });
+    await page.getByText('Contract Abbreviation').click();
+  });
+
+  test('After expand a Functional Group hotkeys for atoms not stop working', async ({
+    page,
+  }) => {
+    /*
+    Test case: EPMLSOPKET-12988
+    Description: After pressing hotkey 'N' it can be placed on canvas.
+    Test working not properly. We have open bug https://github.com/epam/ketcher/issues/2591
+    */
+    const x = 300;
+    const y = 300;
+    await selectFunctionalGroups(FunctionalGroups.Boc, page);
+    await clickInTheMiddleOfTheScreen(page);
+
+    await resetCurrentTool(page);
+    await waitForRender(page, async () => {
```

Comment:
Here `waitForRender` is not required – context menu is not a part of canvas, so it won't wait for "Rendering" of menu. Btw, ` await page.getByText('Expand Abbreviation').click();` will wait for context-menu to appear